### PR TITLE
dui3-175 Direct Shape conversion to native

### PR DESCRIPTION
**Pull request by connorivy**
### Helpers
- ParameterValueSetter
  - Existing utility so translate base object parameters into Revit object parameters without knowing exactly what parameters are being set
- Revit Categories
  - Removed a bunch of code that wasn't being used and added methods to get the revitCategory from an object.
    - I think this is grasshopper specific

### Converters
- All added converters are exising 'ToNative' methods in the current converter that have been slightly modified to remove redundant try catchs and change code styles

## Screenshots:

image
Finished product, rhino meshes in Revit

## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [ ] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [ ] My commits are related to the pull request and do not amend unrelated code or documentation.
- [ ] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

**Review by oguzhankoral — CHANGES_REQUESTED** [review]
For now minor comments, will test in 2h

File: Converters/Revit/Speckle.Converters.RevitShared/Helpers/RevitCategories.cs [hunk 1]
```diff
@@ -12,139 +12,37 @@ namespace Speckle.Converters.RevitShared.Helpers;
...
 {
-  public static Dictionary<string, RevitCategoryInfo> All { get; }
-
-  static RevitCategories()
+  /// <summary>
+  /// Returns the corresponding <see cref="RevitCategory"/> based on a given built-in category name
+  /// </summary>
+  /// <param name="builtInCategory">The name of the built-in category</param>
+  /// <returns>The RevitCategory enum value that corresponds to the given name</returns>
+  public static RevitCategory GetSchemaBuilderCategoryFromBuiltIn(string builtInCategory)
```
[the remark this answers is not in the document]
Comment (connorivy) [reply Converters/Revit/Speckle.Converters.RevitShared/Helpers/RevitCategories.cs:20]: we don't need it as static, but the method has nothing to do with the state in the class, so why not make it static? It saves us from having to make an instance of the Revit Categories class

File: Converters/Revit/Speckle.Converters.RevitShared/Helpers/ParameterValueSetter.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    var param = elem.get_Parameter(bip);
+    if (param != null && !param.IsReadOnly)
+    {
+      param.Set(value ? 1 : 0);
+    }
+  }
+
+  //Shared parameters use a GUID to be uniquely identified
+  //Other parameters use a BuiltInParameter enum
+  private static string GetParamInternalName(DB.Parameter rp)
```
[the remark this answers is not in the document]
Comment (connorivy) [reply Converters/Revit/Speckle.Converters.RevitShared/Helpers/ParameterValueSetter.cs:169]: you're correct that it isn't necessary, but why does it matter? If you don't make it static, then your IDE will give you a suggestion that the method could be made static

[the remark this answers is not in the document]
Comment (oguzhankoral) [reply Converters/Revit/Speckle.Converters.RevitShared/Helpers/ParameterValueSetter.cs:169]: Wasn't it a compromise we took with DI, use static only if it is reaaly needed? cc @BovineOx

[the remark this answers is not in the document]
Comment (adamhathcock) [reply Converters/Revit/Speckle.Converters.RevitShared/Helpers/ParameterValueSetter.cs:169]: static private methods are fine.  It's making things not testable that is the issue.  static private stuff is usually tested by testing something else that is instance based and public

File: Converters/Revit/Speckle.Converters.RevitShared/Helpers/RevitCategories.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamhathcock) [reply Converters/Revit/Speckle.Converters.RevitShared/Helpers/RevitCategories.cs:20]: we want things to be testable.  A user of this static method can't have it's usage mocked away.  It's best to have a constructor injected logic class that can be tested separately.

[the remark this answers is not in the document]
Comment (connorivy) [reply Converters/Revit/Speckle.Converters.RevitShared/Helpers/RevitCategories.cs:20]: these methods are purely utility methods that parse Speckle enums from strings. They don't involve Revit types. Do we need to mock that?

[the remark this answers is not in the document]
Comment (adamhathcock) [reply Converters/Revit/Speckle.Converters.RevitShared/Helpers/RevitCategories.cs:20]: we should test all we can...that usually means Mock.

We should especially test utility methods

**Review by adamhathcock — APPROVED** [review]

**Review by adamhathcock — APPROVED** [review]

**Review by oguzhankoral — APPROVED** [review]

**Pull request merged**